### PR TITLE
chore(react-i18next): change usage of custom useTranslation to react-i18next

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
## Description
Updates out translation code to make use of i18n interpolation instead of sprintf-js. This should help with performance, developer experience and bundle size.

### Todo:

- [x] Remove usage of custom useTranslation hook
- [x] Fix type errors
- [x] Fix tests
- [x] Update translation files
- [x] Remove old custom code

### Double check

- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: web/cypress.config.ts
Comment (VIKTORVAV99) [inline web/cypress.config.ts:29]: This is just so the local setup uses the same as the CI everywhere. Was quite annoying when it picked Swedish instead of English all the time.

File: web/src/features/map-controls/MapOptionSelector.tsx
Comment (VIKTORVAV99) [inline web/src/features/map-controls/MapOptionSelector.tsx:18]: since we consider any variable that begins with `_` to be a placeholder/allowed unused we didn't catch these unused hooks before. 

There are a few more that I removed as well.

File: web/src/App.tsx
Comment (madsnedergaard) [inline web/src/App.tsx:81]: Why not use the `useTranslation` hook here?

File: web/src/features/charts/CarbonChart.tsx
Comment (madsnedergaard) [inline web/src/features/charts/CarbonChart.tsx:28]: Nice catch :)

File: web/src/features/charts/graphUtils.ts
Comment (madsnedergaard) [inline web/src/features/charts/graphUtils.ts:164]: Nice with this change so we don't need the hook in here :)

File: web/src/features/charts/tooltips/AreaGraphTooltipHeader.tsx
Comment (madsnedergaard) [inline web/src/features/charts/tooltips/AreaGraphTooltipHeader.tsx:49]: It's weird how we were mixing `i18n.t` and `__`, so it's great to get it all aligned 😅

File: web/src/features/charts/tooltips/BreakdownChartTooltip.tsx
Comment (madsnedergaard) [inline web/src/features/charts/tooltips/BreakdownChartTooltip.tsx:160]: Omg yes, I have many times struggled with trying to reason about what the hell this exact translation would output - the named variables makes it so much easier 😍

**Review by madsnedergaard — APPROVED** [review]
My fingers hurt from scrolling, but it all looks great and on the preview app everything looks as expected! Thanks for taking the time to do this one, it's cleaning up a lot of tech debt 🙏 💚

File: web/src/App.tsx
Comment (VIKTORVAV99) [reply web/src/App.tsx:81]: Good question 😅

I'll change it...

File: web/src/features/charts/tooltips/AreaGraphTooltipHeader.tsx
Comment (VIKTORVAV99) [reply web/src/features/charts/tooltips/AreaGraphTooltipHeader.tsx:49]: That was me like last week, got so annoyed by sprintf I ended up removing it everywhere 😅

File: web/src/features/charts/tooltips/BreakdownChartTooltip.tsx
Comment (VIKTORVAV99) [reply web/src/features/charts/tooltips/BreakdownChartTooltip.tsx:160]: And now you know why this PR exists 😅

File: web/src/App.tsx
Comment (VIKTORVAV99) [reply web/src/App.tsx:81]: I remember why I did it like this now...

Otherwise the i18n import becomes unused and well vite removes the whole thing 😅

It should still be the same instance though so should not impact any functionality.

**Comment by VIKTORVAV99** [comment]
@madsnedergaard should we wait for a review from Tony or merge this now? I would like to merge this as soon as possible since it changes 88 files and merge conflicts are only a matter of time...

File: web/src/App.tsx
Comment (madsnedergaard) [reply web/src/App.tsx:81]: Ah okay, fair :)

**Pull request merged**
